catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/dialog/LegoSensorPortConfigDialogTest.java

```diff
@@ -0,0 +1,127 @@
+import static org.catrobat.catroid.uiespresso.ui.fragment.rvutils.RecyclerViewInteractionWrapper.onRecyclerView;
+
+@Category({Cat.AppUi.class, Level.Smoke.class})
+public class LegoSensorPortConfigDialogTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<SpriteActivity> baseActivityTestRule = new
+					BaseActivityInstrumentationRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	private NXTSensor.Sensor[] sensorMapping;
+	private boolean legoNXTBrickSetting;
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("LegoSensorPortConfigDialogTest");
+		script.addBrick(new ChangeSizeByNBrick(0));
+
+		legoNXTBrickSetting = getNXTBrickSetting();
+		setNXTBrickSetting(true);
+
+		sensorMapping = SettingsFragment.getLegoNXTSensorMapping(InstrumentationRegistry.getTargetContext());
+
+		SettingsFragment.setLegoMindstormsNXTSensorMapping(InstrumentationRegistry.getTargetContext(),
+						new NXTSensor.Sensor[] {NXTSensor.Sensor.NO_SENSOR, NXTSensor.Sensor.NO_SENSOR,
+								NXTSensor.Sensor.NO_SENSOR, NXTSensor.Sensor.NO_SENSOR});
+
+		baseActivityTestRule.launchActivity();
+
+		onBrickAtPosition(1).onChildView(withId(R.id.brick_change_size_by_edit_text)).perform(click());
+
+		openDialog();
+	}
+
+	@Test
+	public void pressOKTest() {
+		onLegoSensorPortConfigDialog().performClickOnOK();
+		onLegoSensorPortConfigDialog();
+	}
+
+	@Test
+	public void checkDialogTest() {
+		onLegoSensorPortConfigDialog().performClickOnPort(1, R.string.nxt_no_sensor);
+		onLegoSensorPortConfigDialog().performClickOnOK();
+		Espresso.onView(withText(R.string.formula_editor_sensor_lego_nxt_1));
+
+		openDialog();
+
+		onLegoSensorPortConfigDialog().checkPortDisplayed(1, R.string.nxt_sensor_touch);
+	}
+
+	@After
+	public void tearDown() {
+		SettingsFragment.setLegoMindstormsNXTSensorMapping(InstrumentationRegistry.getTargetContext(), sensorMapping);
+		setNXTBrickSetting(legoNXTBrickSetting);
+	}
+
+	private void setNXTBrickSetting(boolean bricksEnabled) {
+		SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext()).edit();
+		editor.putBoolean(SettingsFragment.SETTINGS_MINDSTORMS_NXT_BRICKS_ENABLED, bricksEnabled).commit();
+	}
+
+	private boolean getNXTBrickSetting() {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+		return sharedPreferences.getBoolean(SettingsFragment.SETTINGS_MINDSTORMS_NXT_BRICKS_ENABLED, false);
+	}
+
+	private void openDialog() {
+		onFormulaEditor().performOpenCategory(FormulaEditorWrapper.Category.DEVICE);
+		String formulaEditorSensorString = UiTestUtils.getResourcesString(R.string.formula_editor_sensor_lego_nxt_touch);
+		onRecyclerView().performOnItemWithText(formulaEditorSensorString, click());
```

Comment:
please use FormulaEditorCategoryListWrapper.performSelect
please dont resolve the string there, thers already a method for selecting stuff there based on stringId

you can even go there and do
onFormulaEditor()
     .performOpenCategory(FormulaEditorWrapper.Category.DEVICE);
     .performSelect(R.string.formula_editor_sensor_lego_nxt_touch);

Reply:
The FormulaCategoryListWrapper performSelect does not work, because onFormulaEditor is called after performSelect and onFormulaEditor checks if the current View is the Formula Editor.
Also the same for the second option with onFormulaEditor.

Reply:
then check usages, if it is actually required anywhere, if not remove the onFormulaEditor at the end, and change return type to void